GAE/src/org/waterforpeople/mapping/dataexport/RawDataSpreadsheetImporter.java

```diff
@@ -120,19 +124,38 @@ public void executeImport(File file, String serverBase, Map<String, String> crit
         try {
             log.info(String.format("Importing %s to %s using criteria %s", file, serverBase,
                     criteria));
-            Sheet sheet = getDataSheet(file);
-            String surveyId = criteria.get("surveyId");
-            Map<Integer, Long> columnIndexToQuestionId = processHeader(sheet);
+            Workbook wb = getWorkbook(file);
+            
+            //Find out if this is a 2017-style report w group headers and rqg's on separate sheets
+            splitSheets = safeCellCompare(wb.getSheetAt(0), 0, 0, METADATA_HEADER);
+            int headerRowIndex = splitSheets ? 1 : 0;
+     
+            Map<Sheet, Map<Integer, Long>> sheetMap = new HashMap<>();
+            // Find all data sheets
+            for (int i = 0; i < wb.getNumberOfSheets(); i++) {
+                Sheet sheet = wb.getSheetAt(i); 
+                String sn = sheet.getSheetName();
+                if (i == 0 || sn.startsWith("Group ")) {
+                    sheetMap.put(sheet, processHeader(sheet, headerRowIndex));
+                }
+            }
+            
             Map<Long, QuestionDto> questionIdToQuestionDto = fetchQuestions(serverBase, criteria);
 
             Map<Long, List<QuestionOptionDto>> optionNodes = fetchOptionNodes(serverBase,
                     criteria, questionIdToQuestionDto.values());
 
-            List<InstanceData> instanceDataList = parseSheet(sheet, questionIdToQuestionDto,
-                    columnIndexToQuestionId, optionNodes);
-
+            List<InstanceData> instanceDataList = new ArrayList<>();
+            if (splitSheets) {
+                instanceDataList = parseSplitSheets(wb.getSheetAt(0), sheetMap, questionIdToQuestionDto,
```

Comment:
Looking into `parseSplitSheets` and `parseSingleSheet`, they are quite similar. Is it possible to keep a generic function that is `parseSheet` to which you will pass either base or RQG sheet? It may require making multiple requests for data upload per instance but thats something we can fix after a working MVP.

Reply:
I was thinking we would remove the single-sheet flavor once people are happy with the new format and no longer have old stuff to import. Thus it was sub-optimal to spend time to figure out exactly what could be "deduplicated".

Reply:
OK. In that case please make an issue for it and we can follow up in next releases.

Reply:
I was preparing for how it will work for *new* data in split sheets. The user will have to put a special value like new-1, new-2 in the "Identifier" column on both the base and RQG sheet, and when they have been collected into the same instance that is blanked out and when uploaded they will get a new identifier assigned.

Reply:
Added issue #2230 for removing parseSingleSheets() after a transition period.